src/io/obj.jl

```diff
@@ -126,4 +136,26 @@ end
 
 function _typemax(::Type{OffsetInteger{O, T}}) where {O, T}
     typemax(T)
+end
+
+function save(f::Stream{format"OBJ"}, mesh::AbstractMesh)
+    io = stream(f)
+    for p in decompose(Point3f0, mesh)
+        println(io, "v ", p[1], " ", p[2], " ", p[3])
+    end
+    
+    # TODO
+    # How do you check if normals and uvs are available? decompose generates 
+    # them, but we shouldn't save them if they're not in the mesh
+    for uv in decompose_uv(mesh)
+        println(io, "vt ", uv[1], " ", uv[2])
+    end
+
+    for n in decompose_normals(mesh)
+        println(io, "vn ", n[1], " ", n[2], " ", n[3])
+    end
```

Comment:
It would be better to have 
```
if has(mesh, :normals)
    for n in mesh.normals
        ...
```
so that we don't create extra data. Does something like `has` exist in GeometryBasics?

Reply:
Ah, `hasproperty`